crates/bevy_input_focus/src/directional_navigation.rs

```diff
@@ -259,8 +363,323 @@ pub enum DirectionalNavigationError {
     },
 }
 
+/// A navigation node with position and size information.
+///
+/// This struct represents a UI element in the automatic directional navigation system,
+/// containing its entity ID, center position, and size for spatial navigation calculations.
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[cfg_attr(
+    feature = "bevy_reflect",
+    derive(Reflect),
+    reflect(Debug, PartialEq, Clone)
+)]
+pub struct NavigationNode {
+    /// The entity identifier for this navigation node.
+    pub entity: Entity,
+    /// The center position of the node in global coordinates.
+    pub position: Vec2,
+    /// The size (width, height) of the node.
+    pub size: Vec2,
+}
+
+/// Trait for extracting position and size from UI node components.
+///
+/// This allows the auto-navigation system to work with different UI implementations
+/// as long as they can provide position and size information.
+pub trait NavigableNode {
+    /// Returns the center position and size of the node in global coordinates.
+    fn get_bounds(&self) -> (Vec2, Vec2);
+}
+
+// We can't directly implement this for `bevy_ui` types here without circular dependencies,
+// so we'll use a more generic approach with separate functions for different component sets.
+
+/// Convert `CompassOctant` to a unit direction vector.
+///
+fn octant_to_direction(octant: CompassOctant) -> Vec2 {
```

Comment:
This doesn't need to exist: you should use the `From<CompassOctant> for Dir2` that already exists.